docs/template_plugin/src/template_pattern_transformation.cpp

```diff
@@ -43,9 +45,94 @@ void ngraph::pass::MyPatternBasedTransformation::transform() {
         return true;
     };
 
-    // Register pattern with divide operaiton as a pattern root node
+    // Register pattern with Divide operation as a pattern root node
     auto m = std::make_shared<ngraph::pattern::Matcher>(div, "ConvertDivide");
     // Register Matcher
-    this->add_matcher(m, callback, ngraph::pass::PassProperty::CHANGE_DYNAMIC_STATE);
+    this->register_matcher(m, callback);
 }
 // ! [graph_rewrite:template_transformation_cpp]
+
+// ! [matcher_pass:relu_fusion]
+ngraph::pass::ReluReluFusionMatcher::ReluReluFusionMatcher() {
+    auto m_relu1 = ngraph::pattern::wrap_type<ngraph::opset3::Relu>(pattern::consumers_count(1));
+    auto m_relu2 = ngraph::pattern::wrap_type<ngraph::opset3::Relu>({m_relu1});
+
+    ngraph::matcher_pass_callback callback = [=](pattern::Matcher& m) {
+        // Map that helps to connect labels with matched outputs
+        auto& node_to_output = m.get_pattern_value_map();
+
+        // Create new Relu operation and add register it for additional execution
+        auto new_relu = register_new_node<ngraph::opset3::Relu>(
+            node_to_output.at(m_relu1).get_node_shared_ptr()->input_value(0));
+
+        // Copy runtime info attributes to newly created operation
+        ngraph::copy_runtime_info(m.get_matched_nodes(), new_relu);
+
+        // Save last Relu name to new Relu operation
+        new_relu->set_friendly_name(m.get_match_root()->get_friendly_name());
+
+        // Replace Relu->Relu with Relu
+        ngraph::replace_node(m.get_match_root(), new_relu);
+
+        // Return true as the root node was changed
+        return true;
+    };
+
+    // Register pattern with Relu operation as a pattern root node
+    auto m = std::make_shared<ngraph::pattern::Matcher>(m_relu2, "ReluReluFusion");
+    // Register Matcher
+    this->register_matcher(m, callback);
+}
+// ! [matcher_pass:relu_fusion]
+
+void run_matcher_on_node(std::shared_ptr<ngraph::Node> node) {
```

Comment:
reference to shared_ptr

Reply:
Will be used only in docs and this function declaration won't be included.